### PR TITLE
Package script for coverage, increase tests performance

### PR DESCRIPTION
**Pull request by standy**
That PR mostly about tests and typescript configuration. I've done root `tsconfig` and `jest.config` with project paths, so tests do not require build anymore. 
And I have replaced `compileTs` method in tests with `validateTs`+`toMatchSnapshot`, compile ts is too slow for tests (about 10s each) and not so useful, since `Cannot find module` error is ignored  
All other changes are just prettier and typecheck fixes

Improvements list:
- [x] Script to run all tests at once. No build required for tests
- [x] Increasing tests speed (5m -> 30s)
- [x] Code coverage
- [x] Script to run prettier for all files
- [x] Script for checking types
- [x] Faster cleanup script for `node_modules`
- [x] Removed few old compiled files in `dev-test`, they were unused

I hope it will increase the development experience. Besides that, always useful to know your test coverage ✌️

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -1,8 +1,6 @@
 {
   "trailingComma": "es5",
-  "tabWidth": 2,
```
Comment (standy) [inline .prettierrc]: This is default

File: packages/graphql-codegen-cli/tests/cli-error.spec.ts [hunk 2]
```diff
@@ -2,14 +2,18 @@ import { cliError } from '../src/utils/cli-error';
 
 describe('cliError', () => {
   let spyProcessExit: jest.SpyInstance;
+  let spyConsoleError: jest.SpyInstance;
 
   beforeEach(() => {
     spyProcessExit = jest.spyOn(process, 'exit');
     spyProcessExit.mockImplementation();
+    spyConsoleError = jest.spyOn(console, 'error');
```
Comment (standy) [inline packages/graphql-codegen-cli/tests/cli-error.spec.ts]: Silence console while testing

File: packages/plugins/flow/operations/tests/flow-documents.spec.ts [hunk 3]
```diff
@@ -3,12 +3,9 @@ import '@graphql-codegen/testing';
 import { parse, buildClientSchema, buildSchema } from 'graphql';
 import { plugin } from '../src/index';
 import { validateFlow } from '../../flow/tests/validate-flow';
-import { readFileSync } from 'fs';
 
 describe('Flow Operations Plugin', () => {
-  const gitHuntSchema = buildClientSchema(
-    JSON.parse(readFileSync('../../../../dev-test/githunt/schema.json', 'utf-8'))
-  );
+  const gitHuntSchema = buildClientSchema(require('../../../../../dev-test/githunt/schema.json'));
```
Comment (standy) [inline packages/plugins/flow/operations/tests/flow-documents.spec.ts]: `require` is relative to the current file, while `readFileSync` is a relative to `process.cwd()`

File: packages/plugins/java/resolvers/src/visitor.ts [hunk 4]
```diff
@@ -32,7 +32,7 @@ export interface JavaResolverParsedConfig extends ParsedConfig {
 export class JavaResolversVisitor extends BaseVisitor<JavaResolversPluginRawConfig, JavaResolverParsedConfig> {
   private _includeTypeResolverImport = false;
 
-  constructor(rawConfig: JavaResolversPluginRawConfig, private _schema: GraphQLSchema, defaultPackageName: string) {
+  constructor(rawConfig: JavaResolversPluginRawConfig, _schema: GraphQLSchema, defaultPackageName: string) {
```
Comment (standy) [inline packages/plugins/java/resolvers/src/visitor.ts]: Typescript warning, unused field

File: packages/plugins/typescript/type-graphql/src/index.ts [hunk 5]
```diff
@@ -35,4 +35,4 @@ export const plugin: PluginFunction<TypeGraphQLPluginConfig> = (
   };
 };
 
-export { TypeGraphQLVisitor, TsIntrospectionVisitor };
+export { TsIntrospectionVisitor };
```
Comment (standy) [inline packages/plugins/typescript/type-graphql/src/index.ts]: `TypeGraphQLVisitor` exported twice

**Comment by dotansimha** [comment]
THIS IS AWESOME!!! Thank you @standy !!! 
I think @kamilkisiela wanted (and started?) to do something similar. 
@kamilkisiela can you please review this?

**Comment by kamilkisiela** [comment]
Yes, I did that locally, wanted to finish it today :) I'm happy someone have done it before, free Sunday then.

I'm reviewing it now!

**Comment by kamilkisiela** [comment]
@dotansimha I say let's merge it and I will add few more things I already did after that

**Comment by standy** [comment]
Thank you! Didn't know you worked on it, @kamilkisiela 
I have some ideas about how to speed up the build step too. Have a plans to make PR in a couple of days, hope you'll enjoy it :)

**Comment by kamilkisiela** [comment]
@standy I want to do `tsc --incremental` that outputs all projects to `<root>/dist` (esmodules + es2015 target) and then run rollup to do cjs and esm bundles in parallel, for all projects.

**Comment by kamilkisiela** [comment]
`tsc` takes 11s and parallel build should take much much less.

**Pull request merged**